openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,212 @@
+    convert_colorspace_for_input_paths,
+    get_transcode_temp_directory,
+)
+
+from openpype.lib.profiles_filtering import filter_profiles
+
+
+class ExtractColorTranscode(publish.Extractor):
+    """
+    Extractor to convert colors from one colorspace to different.
+
+    Expects "colorspaceData" on representation. This dictionary is collected
+    previously and denotes that representation files should be converted.
+    This dict contains source colorspace information, collected by hosts.
+
+    Target colorspace is selected by profiles in the Settings, based on:
+    - families
+    - host
+    - task types
+    - task names
+    - subset names
+    """
+
+    label = "Transcode color spaces"
+    order = pyblish.api.ExtractorOrder + 0.019
+
+    optional = True
+
+    # Supported extensions
+    supported_exts = ["exr", "jpg", "jpeg", "png", "dpx"]
+
+    # Configurable by Settings
+    profiles = None
+    options = None
+
+    def process(self, instance):
+        if not self.profiles:
+            self.log.debug("No profiles present for color transcode")
+            return
+
+        if "representations" not in instance.data:
+            self.log.debug("No representations, skipping.")
+            return
+
+        if not is_oiio_supported():
+            self.log.warning("OIIO not supported, no transcoding possible.")
+            return
+
+        profile = self._get_profile(instance)
+        if not profile:
+            return
+
+        repres = instance.data.get("representations") or []
+        for idx, repre in enumerate(list(repres)):
+            self.log.debug("repre ({}): `{}`".format(idx + 1, repre["name"]))
+            if not self._repre_is_valid(repre):
+                continue
+
+            colorspace_data = repre["colorspaceData"]
+            source_color_space = colorspace_data["colorspace"]
+            config_path = colorspace_data.get("configData", {}).get("path")
+            if not os.path.exists(config_path):
+                self.log.warning("Config file doesn't exist, skipping")
+                continue
+
+            repre = self._handle_original_repre(repre, profile)
+
+            for _, output_def in profile.get("outputs", {}).items():
+                new_repre = copy.deepcopy(repre)
+
+                new_staging_dir = get_transcode_temp_directory()
+                original_staging_dir = new_repre["stagingDir"]
+                new_repre["stagingDir"] = new_staging_dir
+                files_to_convert = new_repre["files"]
+                if not isinstance(files_to_convert, list):
+                    files_to_convert = [files_to_convert]
+
+                files_to_delete = copy.deepcopy(files_to_convert)
+
+                output_extension = output_def["output_extension"]
+                output_extension = output_extension.replace('.', '')
+                if output_extension:
+                    new_repre["name"] = output_extension
+                    new_repre["ext"] = output_extension
```

Comment:
In some cases a representation `name` might not match exactly the `ext` - e.g. for burnins or alike. I think this is currently an oversimplification if it'd ever run over a representation whose name doesn't equal exactly the representation. Plus, there is an off chance the other representation we're changing too already existing in the instance's representation, if e.g. a particular instance would already be generating a `exr` and a `png` format.

Reply:
This would be probably better to mimic what Exctract review does. So the name of representation is the inherited form the output key - it is at the moment ignored with `_` at iteration. Here is what I mean:
![image](https://user-images.githubusercontent.com/40640033/213545372-a56f1f16-64d1-4e58-8a08-c5b27acc00c4.png)

Reply:
> This would be probably better to mimic what Exctract review does. So the name of representation is the inherited form the output key - it is at the moment ignored with `_` at iteration. Here is what I mean: ![image](https://user-images.githubusercontent.com/40640033/213545372-a56f1f16-64d1-4e58-8a08-c5b27acc00c4.png)

@kalisp this hasn't been implemented, are you planning to add it? I would rather use the name as it will come handy. Here is an examle. I am having two output profiles with the same extentions but different configurations. The name allwos me to differentiate them. With current workfow this is not able to work. I would like the loader offer me two different representations.

Reply:
I dont think its a good workflow to have implicit functionality through which name we use for the output definitions. If there is some functionality that an output definition needs, it should have a toggle in the settings.